### DISCUSSION TITLE
Modify Transpiler/Add `"type": "module"` to `package.json`

### DISCUSSION THREAD

**Discussion by matthewoestreich** [material-table-core/core#289] [Ideas]
Hey @Domino987 I wanted to get your thoughts on this..

Currently, we use Babel to transpile our code (comes out to ~370kb). How would you feel about using `esbuild` to transpile? The output is about half the size at ~160kb.

Doing so would also mean we would have to specify `"type": "module"` in our package.json file. This is because others need to know that `import`/`export` **must** be used when using our library. Which shouldn't be an issue because it's a React library, which uses ES6 features by default (and the consumer is already responsible for transpiling the code further if they wish).

Thoughts? 

I have already been testing this and it appears I have resolved all prior issues.

**Comment by Domino987** [comment]
I am not really used to the whole module thing to be honest. I just use CRA. So I can barely say something about it. Why not release a major version and see if problems arise?

**Comment by Domino987** [comment]
Quite a drastic reducing even on [bundlephobia](https://bundlephobia.com/package/@material-table/core@4.0.0)

**Reply by matthewoestreich** [reply]
I've got a question for you. Why does bundlephobia show different size than npm? That confuses me lol
